src/node_crypto.cc

```diff
@@ -4038,12 +4071,60 @@ void Sign::SignUpdate(const FunctionCallbackInfo<Value>& args) {
   sign->CheckThrow(err);
 }
 
+static int Node_SignFinal(EVP_MD_CTX *mdctx, unsigned char *md, unsigned int *s,
+                          EVP_PKEY *pkey, int padding, int pss_saltlen) {
+  unsigned char m[EVP_MAX_MD_SIZE];
+  unsigned int m_len;
+  int i = 0;
+  EVP_PKEY_CTX *pkctx = nullptr;
+
+  *s = 0;
+  if (!EVP_DigestFinal_ex(mdctx, m, &m_len))
+    return i;
+
+  if (mdctx->digest->flags & EVP_MD_FLAG_PKEY_METHOD_SIGNATURE) {
+    size_t sltmp = static_cast<size_t>(EVP_PKEY_size(pkey));
+    i = 0;
+    pkctx = EVP_PKEY_CTX_new(pkey, nullptr);
+    if (!pkctx)
+      goto err;
+    if (EVP_PKEY_sign_init(pkctx) <= 0)
+      goto err;
+    if (pkey->type == EVP_PKEY_RSA || pkey->type == EVP_PKEY_RSA2) {
+      if (EVP_PKEY_CTX_set_rsa_padding(pkctx, padding) <= 0)
+        goto err;
+      if (padding == RSA_PKCS1_PSS_PADDING) {
+        if (EVP_PKEY_CTX_set_rsa_pss_saltlen(pkctx, pss_saltlen) <= 0)
+          goto err;
+      }
+    }
+    if (EVP_PKEY_CTX_set_signature_md(pkctx, mdctx->digest) <= 0)
+      goto err;
+    if (EVP_PKEY_sign(pkctx, md, &sltmp, m, m_len) <= 0)
+      goto err;
+    *s = sltmp;
+    i = 1;
+   err:
+    EVP_PKEY_CTX_free(pkctx);
+    return i;
+  }
+
+  if (!mdctx->digest->sign) {
+    EVPerr(EVP_F_EVP_SIGNFINAL, EVP_R_NO_SIGN_FUNCTION_CONFIGURED);
```

Comment:
As far as I know, this can occur if a hash algorithm is used that does not support signing. When signing with RSA etc., this code will not even be reached (`EVP_MD_FLAG_PKEY_METHOD_SIGNATURE` means "Digest uses EVP_PKEY_METHOD for signing instead of MD specific signing").